src/main/webapp/dev/js/common/assert.es6

```diff
@@ -0,0 +1,19 @@
+class AssertionFailedError extends Error {
+}
+
+function assert(condition, message) {
+    const msg = message === undefined || message === null ? 'Assertion Failed' : message;
+    if (!condition) {
+        throw new AssertionFailedError(msg);
```

Comment:
Not sure why default parameters are not preferred but `||` does not seem to be the only dominant solution? Checking for undefined seems to be widely accepted too, e.g. https://stackoverflow.com/questions/894860/set-a-default-parameter-value-for-a-javascript-function.

Well unless maybe the implementation for default parameters is considered to be a mistake, e.g. not extending to null. I can't find any evidence that it is a mistake though.

It seems both  `default parameters` or `||` have potential issues but it seems neither are substantiated with facts that they are problematic :

**Default Parameters**
`assert(test != null, validator.getErrorMessage()); `
Suppose validator returns `null`, then this does not work as we expect. Though should we consider `null` as a proper return value? Both `Typescript` and `Flow` both support non-nullable types but they seem to have differing opinions on it.

**|| Operator**
Extends to any value that can be interpreted as `false`, in theory, this sounds scary but not sure whether this is a problem in practice.

But default parameters are built into the language itself and can be more concise, is there any reason for us to use our own implementation if we can't find real faults with default parameters?